### PR TITLE
Fixup compute data commitment (#109) and cleaner cid types

### LINKED ISSUE #109: storage market actor should not send piece infos across method call boundaries
I had this discussion with the specs team a while ago, but it seems it was missed. 'GetPieceInfosForDealIDs' should be changed to 'ComputeDataCommitment' (https://github.com/filecoin-project/lotus/blob/master/chain/actors/actor_storagemarket.go#L614) Otherwise we end up having to serialize large numbers of piece infos across method call boundaries, leading to unnecessarily expensive actor calls. 

cc @magik6k

### PR DESCRIPTION
**Pull request by whyrusleeping**
Should resolve #109, also moves us closer to integration success.

### CODE REVIEW

File: actors/abi/sector.go [hunk 1]
```diff
@@ -1,9 +1,8 @@
 package abi
 
 import (
-	cid "github.com/ipfs/go-cid"
-
 	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	cid "github.com/ipfs/go-cid"
```
Comment (anorth) [inline actors/abi/sector.go]: Nit: Please follow the import ordering/grouping of `goimports -local` (or at least don't regress it).

File: actors/abi/sector.go [hunk 2]
```diff
@@ -35,9 +34,6 @@ func NewStoragePower(n int64) StoragePower {
 	return StoragePower(big.NewInt(n))
 }
 
-type UnsealedSectorCID cid.Cid // CommD
-type SealedSectorCID cid.Cid   // CommR
-
```
Comment (anorth) [inline actors/abi/sector.go]: Thanks, I have the same thing in an in-flight PR

File: actors/abi/sector.go [hunk 3]
```diff
@@ -94,10 +90,10 @@ type PartialTicket []byte // 32 bytes
...
 type PoStVerifyInfo struct {
 	Randomness      PoStRandomness
-	CommR           SealedSectorCID
-	Candidates      []PoStCandidate              // From OnChain*PoStVerifyInfo
-	Proofs          []PoStProof                  // From OnChain*PoStVerifyInfo
-	EligibleSectors map[SectorID]SealedSectorCID // TODO: HAMT?
+	CommR           cid.Cid
+	Candidates      []PoStCandidate      // From OnChain*PoStVerifyInfo
+	Proofs          []PoStProof          // From OnChain*PoStVerifyInfo
+	EligibleSectors map[SectorID]cid.Cid // TODO: HAMT?
```
Comment (anorth) [inline actors/abi/sector.go]: While we're here, what do you propose for this data structure. HAMT makes no sense here. How about a list of tuples?

File: actors/builtin/storage_market/storage_market_actor.go [hunk 4]
```diff
@@ -228,31 +231,43 @@ func (a StorageMarketActor) VerifyDealsOnSectorProveCommit(rt Runtime, params *V
...
 }
 
-type GetPieceInfosForDealIDsParams struct {
-	DealIDs []abi.DealID
+type ComputeDataCommitmentParams struct {
+	DealIDs    []abi.DealID
+	SectorSize abi.SectorSize
+}
+
+// cbor gen and delete me
```
Comment (anorth) [inline actors/builtin/storage_market/storage_market_actor.go]: Don't forget!

**Comment by whyrusleeping** [comment]
@anorth thank you for fixing up those things for me!

**Pull request merged**